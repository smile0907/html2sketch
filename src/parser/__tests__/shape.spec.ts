import { isUpdate, outputJSONData, shapeJSON } from '@test-utils';
import { parseToShape } from 'html2sketch';

describe('parseToShape', () => {
  afterAll(() => {
    document.body.innerHTML = '';
  });
  beforeAll(() => {
    document.body.innerHTML = `<div id="shape" style="background: aqua;width:200px"></div>`;
  });
  it('文本正常解析', () => {
    const node = document.getElementById('shape') as HTMLDivElement;

    const shape = parseToShape(node);
    expect(shape).toBeTruthy();
    expect(shape.toSketchJSON()).toMatchSnapshot();
    if (isUpdate) {
      // 如果出现小数点的不一致 进行重新输出
      outputJSONData(shape.toSketchJSON(), 'shape');
    }

    expect(shape.toSketchJSON()).toStrictEqual(shapeJSON);
  });
});
