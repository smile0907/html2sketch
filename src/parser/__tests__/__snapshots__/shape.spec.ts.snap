// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseToShape 文本正常解析 1`] = `
Object {
  "_class": "rectangle",
  "booleanOperation": -1,
  "clippingMaskMode": 0,
  "do_objectID": "UUID",
  "edited": false,
  "exportOptions": Object {
    "_class": "exportOptions",
    "exportFormats": Array [],
    "includedLayerIds": Array [],
    "layerOptions": 0,
    "shouldTrim": false,
  },
  "fixedRadius": 0,
  "frame": Object {
    "_class": "rect",
    "constrainProportions": false,
    "height": 0,
    "width": 200,
    "x": 8,
    "y": 8,
  },
  "hasConvertedToNewRoundCorners": true,
  "isClosed": true,
  "isFixedToViewport": false,
  "isFlippedHorizontal": false,
  "isFlippedVertical": false,
  "isLocked": false,
  "isVisible": true,
  "layerListExpandedType": 0,
  "name": "rect",
  "nameIsFixed": false,
  "needsConvertionToNewRoundCorners": false,
  "pointRadiusBehaviour": 1,
  "points": Array [
    Object {
      "_class": "curvePoint",
      "cornerRadius": 0,
      "curveFrom": "{0, 0}",
      "curveMode": 1,
      "curveTo": "{0, 0}",
      "hasCurveFrom": false,
      "hasCurveTo": false,
      "point": "{0, 0}",
    },
    Object {
      "_class": "curvePoint",
      "cornerRadius": 0,
      "curveFrom": "{1, 0}",
      "curveMode": 1,
      "curveTo": "{1, 0}",
      "hasCurveFrom": false,
      "hasCurveTo": false,
      "point": "{1, 0}",
    },
    Object {
      "_class": "curvePoint",
      "cornerRadius": 0,
      "curveFrom": "{1, 1}",
      "curveMode": 1,
      "curveTo": "{1, 1}",
      "hasCurveFrom": false,
      "hasCurveTo": false,
      "point": "{1, 1}",
    },
    Object {
      "_class": "curvePoint",
      "cornerRadius": 0,
      "curveFrom": "{0, 1}",
      "curveMode": 1,
      "curveTo": "{0, 1}",
      "hasCurveFrom": false,
      "hasCurveTo": false,
      "point": "{0, 1}",
    },
  ],
  "resizingConstraint": 63,
  "resizingType": 0,
  "rotation": 0,
  "shouldBreakMaskChain": false,
  "style": Object {
    "_class": "style",
    "borderOptions": Object {
      "_class": "borderOptions",
      "dashPattern": Array [],
      "isEnabled": true,
      "lineCapStyle": 0,
      "lineJoinStyle": 0,
    },
    "borders": Array [
      Object {
        "_class": "border",
        "color": Object {
          "_class": "color",
          "alpha": 1,
          "blue": 0,
          "green": 0,
          "red": 0,
        },
        "contextSettings": Object {
          "_class": "graphicsContextSettings",
          "blendMode": 0,
          "opacity": 1,
        },
        "fillType": 0,
        "gradient": Object {
          "_class": "gradient",
          "elipseLength": 0,
          "from": "{0, 0}",
          "gradientType": 0,
          "stops": Array [],
          "to": "{1, 0}",
        },
        "isEnabled": true,
        "position": 1,
        "thickness": 0,
      },
    ],
    "colorControls": Object {
      "_class": "colorControls",
      "brightness": 0,
      "contrast": 1,
      "hue": 0,
      "isEnabled": false,
      "saturation": 1,
    },
    "contextSettings": Object {
      "_class": "graphicsContextSettings",
      "blendMode": 0,
      "opacity": 1,
    },
    "do_objectID": "UUID",
    "endMarkerType": 0,
    "fills": Array [
      Object {
        "_class": "fill",
        "color": Object {
          "_class": "color",
          "alpha": 1,
          "blue": 1,
          "green": 1,
          "red": 0,
        },
        "contextSettings": Object {
          "_class": "graphicsContextSettings",
          "blendMode": 0,
          "opacity": 1,
        },
        "fillType": 0,
        "gradient": Object {
          "_class": "gradient",
          "elipseLength": 0,
          "from": "{0, 0}",
          "gradientType": 0,
          "stops": Array [],
          "to": "{1, 0}",
        },
        "isEnabled": true,
        "noiseIndex": 0,
        "noiseIntensity": 0,
        "patternFillType": 1,
        "patternTileScale": 1,
      },
    ],
    "innerShadows": Array [],
    "miterLimit": 10,
    "shadows": Array [],
    "startMarkerType": 0,
    "windingRule": 1,
  },
}
`;
